poll_appengine.py

```diff
@@ -136,20 +139,28 @@ def run(self):
                 isinstance(self.controller.robot, FakeRobot)):
               logging.info("Placing glass")
               actions[0].force = True
+        self.prepare_queue_for_display(queue)
         self.write(queue)
       except urllib2.URLError, e:
         logging.warning("URLError: %s", e)
       except ValueError, e:
         print e
       time.sleep(self.poll_frequency_secs)
 
+  def prepare_queue_for_display(self, queue):
+    self.queue_for_display = {
+      'finished_drink': self.finished_drink,
+      'current_drink': self.current_drink,
+      'queue': queue or []
+    }
+
   def write(self, queue):
```

Comment:
It's kind of wonky that the queue argument is ignored here -- maybe make queue_for_display a local variable in this function, and inline prepare_queue_for_display here?